pilot/pkg/security/authz/model/generator.go

```diff
@@ -145,39 +147,67 @@ func (srcPrincipalGenerator) principal(key, value string, _ bool, useAuthenticat
 	return principalAuthenticated(m, useAuthenticated), nil
 }
 
-type requestPrincipalGenerator struct{}
+type requestPrincipalGenerator struct {
+	useExtendedJwt bool
+}
 
 func (requestPrincipalGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
 	return nil, fmt.Errorf("unimplemented")
 }
 
-func (requestPrincipalGenerator) principal(key, value string, forTCP bool, _ bool) (*rbacpb.Principal, error) {
+func (rpg requestPrincipalGenerator) principal(key, value string, forTCP bool, _ bool) (*rbacpb.Principal, error) {
 	if forTCP {
 		return nil, fmt.Errorf("%q is HTTP only", key)
 	}
-
+	if rpg.useExtendedJwt {
+		iss, sub, _ := strings.Cut(value, "/")
```

Comment:
we should probably reject if there is no `/` in it here, and add a warning into the webhook if a user does this?

Reply:
Actually it was valid to have something like `a*` which matches iss with prefix `a` and an `sub`. Let me update it.